app/src/main/java/com/devshish/internship/presentation/ui/MainViewModel.kt

```diff
@@ -0,0 +1,79 @@
+
+import android.support.v4.media.MediaBrowserCompat
+import android.support.v4.media.MediaMetadataCompat
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.devshish.internship.domain.model.Song
+import com.devshish.internship.presentation.ui.player.*
+import com.devshish.internship.presentation.ui.utils.*
+
+class MainViewModel(
+    private val musicServiceConnection: MusicServiceConnection
+) : ViewModel() {
+
+    private val _mediaItems = MutableLiveData<Resource<List<Song>>>()
+    val mediaItems: LiveData<Resource<List<Song>>> = _mediaItems
+
+    val isConnected = musicServiceConnection.isConnected
+    val networkError = musicServiceConnection.networkError
+    val curPlayingSong = musicServiceConnection.curPlayingSong
+    val playbackState = musicServiceConnection.playbackState
+
+    init {
+        _mediaItems.postValue(Resource.loading(null))
+        musicServiceConnection.subscribe(Constants.MEDIA_ROOT_ID, object : MediaBrowserCompat.SubscriptionCallback() {
+            override fun onChildrenLoaded(
+                parentId: String,
+                children: MutableList<MediaBrowserCompat.MediaItem>
+            ) {
+                super.onChildrenLoaded(parentId, children)
+                val items = children.map {
+                    Song(
+                        mediaId = it.mediaId!!,
+                        title = it.description.title.toString(),
+                        artist = it.description.subtitle.toString(),
+                        uri = it.description.mediaUri,
+                        imageUri = it.description.iconUri,
+                        duration = it.description.describeContents()
+                    )
+                }
+                _mediaItems.postValue(Resource.success(items))
+            }
+        })
+    }
+
+    fun skipToNextSong() {
+        musicServiceConnection.transportControls.skipToNext()
+    }
+
+    fun skipToPreviousSong() {
+        musicServiceConnection.transportControls.skipToPrevious()
+    }
+
+    fun seekTo(position: Long) {
+        musicServiceConnection.transportControls.seekTo(position)
+    }
+
+    fun playOrToggleSong(mediaItem: Song, toggle: Boolean = false) {
+        val isPrepared = playbackState.value?.isPrepared ?: false
+        if (isPrepared && mediaItem.mediaId ==
+            curPlayingSong.value?.getString(MediaMetadataCompat.METADATA_KEY_MEDIA_ID)) {
+            playbackState.value?.let { playbackState ->
+                when {
+                    playbackState.isPlaying -> if (toggle) musicServiceConnection.transportControls.pause()
+                    playbackState.isPlayEnabled -> musicServiceConnection.transportControls.play()
+                    else -> Unit
+                }
+            }
+        } else {
+            musicServiceConnection.transportControls.playFromMediaId(mediaItem.mediaId, null)
+        }
+    }
```

Comment:
it is not good method. Let's store state of playback and so on in MusicServiceConnection
And it is better to make so that ViewModel will know only about MusticServiceConnection class(your class), and not about concrete implementations of music things.
Здесь перейду на русский язык. Но суть в том, чтобы ты мог в случае чего поменять штуку для произведения музыки внутри MusicServiceConnection, при этом чтобы классы которые  использую MusicServiceConnection вообще не поменялись